indexer/src/main.rs

```diff
@@ -165,16 +188,21 @@ async fn main() -> anyhow::Result<()> {
     let receipts_cache_arc: receipts_cache::ReceiptsCacheArc =
         std::sync::Arc::new(Mutex::new(SizedCache::with_size(100_000)));
 
-    let config: near_lake_framework::LakeConfig = opts.to_lake_config().await;
-    let (sender, stream) = near_lake_framework::streamer(config);
-
-    tokio::spawn(metrics::init_server(opts.port).expect("Failed to start metrics server"));
-
     tracing::info!(
         target: INDEXER_FOR_EXPLORER,
         "Starting Indexer for Explorer (lake)...",
     );
 
+    tokio::spawn(metrics::init_server(opts.port).expect("Failed to start metrics server"));
+
+    if opts.start_options() == &StartOptions::FromGenesis {
+        let genesis_file_path = download_genesis_file(&opts).await?;
+        adapters::genesis::store_genesis_records(&pool, genesis_file_path).await?;
```

Comment:
Would be lovely to drop a log message about the start of storing the genesis records, just for clarity.